File: 2021/elixir/day7.ex
{:ok, line} = File.read("inputs/day7.in")

line =
  line
  |> String.split(",", trim: true)
  |> Enum.map(&String.to_integer/1)
  |> Enum.sort()

{min, max} = Enum.min_max(line)

p1_diff =
  Enum.map(min..max, fn n ->
    Enum.reduce(line, 0, fn i, acc ->
      acc + abs(i - n)
    end)
  end)
  |> Enum.min()

p2_diff =
  Enum.map(min..max, fn n ->
    Enum.reduce(line, 0, fn i, acc ->
      x = abs(i - n)
      acc + div(x * (x + 1), 2)
    end)
  end)
  |> Enum.min()

IO.inspect([p1_diff, p2_diff])

File: 2021/python/day7.py
infile = open('inputs/day7.in', 'r')
lines = infile.read()

numbers = [int(i) for i in lines.split(',')]
numbers.sort()

a, b = min(numbers), max(numbers)

least = (-1, 1000000000)
for i in range(a, b + 1):
    t = sum(abs(j - i) * (abs(j - i) + 1) // 2 for j in numbers)
    if t < least[1]:
        least = (i, t)

print(least)